lib/mongo_client.js

```diff
@@ -0,0 +1,211 @@
+////////////////////////////////////////////////////////////////////////////
```

Comment:
This is a JS-ified copy of https://github.com/realm/realm-js/blob/v10/packages/realm-web/src/services/RemoteMongoDBService.ts with only the RemoteMongoDBCollection type and without the factory functions. The constructor was lightly modified to accept the `functions` object directly rather than constructing it itself.